mitiq/shadows/classical_postprocessing.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (C) Unitary Fund
+# Portions of this code have been adapted from PennyLane's tutorial
+# on Classical Shadows.
+# Original authors: PennyLane developers: Brian Doolittle, Roeland Wiersema
+# Tutorial link: https://pennylane.ai/qml/demos/tutorial_classical_shadows
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+"""Classical post-processing process of classical shadows."""
+
+from typing import Tuple, List, Any
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+import mitiq
+
+# local unitary that applied to the qubits
+phase_z = cirq.S._unitary_().conj()
+hadamard = cirq.H._unitary_()
+identity = cirq.I._unitary_()
+PAULI_MAP = {"X": hadamard, "Y": hadamard @ phase_z, "Z": identity}
+
+
+def classical_snapshot(b_list: List[int], u_list: List[str]) -> NDArray[Any]:
+    r"""
+    Implement a single snapshot state reconstruction.
+
+    Args:
+        b_list: The list of classical outcomes for the snapshot. Here,
+            b = 1 corresponds to :math:`|0\rangle`, and
+            b = -1 corresponds to :math:`|1\rangle`.
+
+        u_list: Array of ("X", "Y", "Z") for the applied
+            Pauli measurement on each qubit.
+
+    Returns:
+        Reconstructed snapshot in terms of nparray, which is not a physical
+        state.
+    """
+
+    # z-basis measurement outcomes
+    b_zero = np.array([[1.0 + 0.0j, 0.0 + 0.0j]])
+    zero_state = b_zero.T @ b_zero
+
+    b_one = np.array([[0.0 + 0.0j, 1.0 + 0.0j]])
+    one_state = b_one.T @ b_one
```

Comment:
I think this is a bit more readable, WDYT?
```suggestion
    zero_state = np.array([[1, 0], [0, 0]], dtype=complex)
    one_state = np.array([[0, 0], [0, 1]], dtype=complex)
```

Reply:
Actually not sure if `dtype=complex` is completely needed since below we are taking the tensor product with complex matrices.